src/dfmpreview/src/extension.ts

```diff
@@ -0,0 +1,273 @@
+        return ViewColumn.One;
+    }
+
+    if (!sideBySide) {
+        return active.viewColumn;
+    }
+
+    switch (active.viewColumn) {
+        case ViewColumn.One:
+            return ViewColumn.Two;
+        case ViewColumn.Two:
+            return ViewColumn.Three;
+    }
+
+    return active.viewColumn;
+}
+
+function showSource(mdUri: Uri) {
+    if (!mdUri) {
+        return commands.executeCommand('workbench.action.navigateBack');
+    }
+
+    const docUri = Uri.parse(mdUri.query);
+
+    for (let editor of window.visibleTextEditors) {
+        if (editor.document.uri.toString() === docUri.toString()) {
+            return window.showTextDocument(editor.document, editor.viewColumn);
+        }
+    }
+
+    return workspace.openTextDocument(docUri).then(doc => {
+        return window.showTextDocument(doc);
+    });
+}
+
+//this class is to call the dfmserver(child_process) by Interprocess communication
+class PreviewCore {
+    private _spawn: child_process.ChildProcess;
+    private _waiting: boolean;
+    public _is_firsttime: boolean;
+
+    constructor(context: ExtensionContext) {
+        let extpath = context.asAbsolutePath('./DfmParse/PreviewCore.exe');
+        this._spawn = child_process.spawn(extpath);
+        this._waiting = false;
+
+        this._spawn.stdout.on('data', function (data) {
+            let tmp = data.toString();
+            let endcharcode = tmp.charCodeAt(tmp.length - 1);
+            if (Is_end && endcharcode == ENDCODE) {
+                previewresult = tmp;
+                provider.update(document_uri);
+            }
+            //the first one if the result been cut
+            else if (Is_end && endcharcode != ENDCODE) {
+                previewresult = tmp;
+                Is_end = false;
+            }
+            //the result be cut and this is not the last one
+            else if (!Is_end && endcharcode != ENDCODE) {
+                previewresult += tmp;
+            }
+            //the result be cut and this is the last one
+            else {
+                previewresult += tmp;
+                Is_end = true;
+                provider.update(document_uri);
+            }
+        });
+
+        this._spawn.stderr.on('data', function (data) {
+            console.log("error " + data + '\n');
+        });
+
+        this._spawn.on('exit', function (code) {
+            console.log('child process exit with code ' + code);
+        });
+    }
+
+    private sendtext() {
+        let rtpath = workspace.rootPath;
+        let editor = window.activeTextEditor;
+        if (!editor) {
+            return;
+        }
+        let doc = editor.document;
+        let docContent = doc.getText();
+        let filename = doc.fileName;
+        let rtpath_length = rtpath.length;
+        let filepath = filename.substr(rtpath_length + 1, filename.length - rtpath_length);
+
+        if (doc.languageId === "markdown") {
+            let num_of_row = docContent.split("\r\n").length;
```

Comment:
or use os.EOL

---
In reply to: [73645261](https://github.com/dotnet/docfx/pull/558#discussion_r73645261) [](ancestors = 73645261)